### PR TITLE
Setup Devops Pipeline for Nest App

### PR DESCRIPTION
**Pull request by nikzaugg**
- [x] Dockerize the application
- [x] Setup action for running tests
- [x] Setup action for Docker build and push to Docker Hub
- [x] Setup action for Docker build and push to Heroku registry
- [x] Setup action for triggering a deployment on Heroku

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: Build the Docker image
+        run: docker build . --tag kwizapp/nest-api:latest
+      - name: Push the Docker image to Docker Hub
+        run: docker push kwizapp/nest-api
+
+  build_heroku:
+    runs-on: ubuntu-latest
+    needs: test
+
```
Comment (rschlaefli) [inline .github/workflows/main.yml]:
```suggestion
    if: github.branch == 'dev' || github.branch == 'master'
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: azure/docker-login@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: Build the Docker image
+        run: docker build . --tag kwizapp/nest-api:latest
+      - name: Push the Docker image to Docker Hub
```
Comment (rschlaefli) [inline .github/workflows/main.yml]:
```suggestion
      - name: Push the Docker image to Docker Hub
        if: github.branch == 'dev' || github.branch == 'master'
```

Comment (rschlaefli) [reply .github/workflows/main.yml]: the latest image on docker hub should always be the latest state from our dev branch. we want to run a docker build for every commit to ensure that images can be built. however, we only want to override the docker hub image if we are on dev or master.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        run: docker push kwizapp/nest-api
+
+  build_heroku:
+    runs-on: ubuntu-latest
+    needs: test
```
Comment (rschlaefli) [inline .github/workflows/main.yml]:
```suggestion
    needs: test
    if: github.branch == 'dev' || github.branch == 'master'
```

File: .github/workflows/main.yml (hunk 2 above)
Comment (nikzaugg) [reply .github/workflows/main.yml]: good catch! thanks for that :)

**Pull request merged**